DM-44651: Fix SDSS failed location translation and add more test detail on failure

timj opened the pull request.

timj committed:
Refresh pre-commit

timj committed:
Use uv pip --system

timj committed:
Fallback to using hard-coded APO location if Astropy search fails

timj committed:
Improve detail in error reporting for tests

erykoff commented on python/astro_metadata_translator/translators/sdss.py:
```diff
@@ -183,8 +189,13 @@ def to_location(self) -> EarthLocation:
+        # is not accessible. If it fails fall back to hard-coded location.
+        try:
+            value = EarthLocation.of_site("apo")
+        except astropy.coordinates.UnknownSiteException:
```
You either need to import `astropy.coordinates` or `from astropy.coordinates import UnknownSiteException` or this won't work (I checked by deleting my cache and running without internet).

timj replied:
wow. How come none of the linters complained? VSCode was more than happy to add this...

timj committed:
Fallback to hard-coded locations for DECam and CFHT

Should only trigger on small fraction of datasets combined with
a bad network connection.
